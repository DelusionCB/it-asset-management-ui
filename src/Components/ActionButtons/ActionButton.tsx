import React from 'react'
import {Button} from 'reactstrap';
import {useTranslation} from 'react-i18next';
import './index.scss'
import {postData} from '../../Api/Utils/archiveUtils';
import {useNavigate} from 'react-router-dom'
import {actionButtonPropTypes} from '../../Types/types.components';

function ActionButton ({values, action, type}: actionButtonPropTypes): JSX.Element {
    const {t} = useTranslation()
    const navigate = useNavigate()

    function handleAction (): void {
        switch (action) {
            case 'create' || 'edit':
                void postData(navigate, action, type, values)
                break;
            case 'copy':

                break;
        }
    }

    return (
        <Button
            className='action-button'
            onClick={() => { handleAction(); }}
        >
            {t(`${type}.${action}`)}
        </Button>
    )
}

export default ActionButton
