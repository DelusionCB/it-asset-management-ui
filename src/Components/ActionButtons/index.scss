.action-button {
    background-color: var(--primary-color);
    border-radius: 0;
    font-size: 1.25rem;
    border: 2px solid black;
        &:hover, &:focus {
            border: 2px solid var(--secondary-color);
            background-color: var(--primary-color);
        }
    }
