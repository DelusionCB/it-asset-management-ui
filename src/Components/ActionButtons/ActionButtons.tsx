import React from 'react'
import ActionButton from './ActionButton';
import {ActionPropTypes} from '../../Types/types.components';

function ActionButtons ({values, actions, type}: ActionPropTypes): JSX.Element {
    function getButtons (): JSX.Element[] {
        return actions.map((action, key) => {
            return (
                <ActionButton
                    values={values}
                    action={action}
                    key={key}
                    type={type}
                />
            )
        })
    }

    return (
        <div>
            {getButtons()}
        </div>
    )
}

export default ActionButtons
