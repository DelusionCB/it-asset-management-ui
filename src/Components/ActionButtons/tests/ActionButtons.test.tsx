import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ActionButtons from '../ActionButtons';
import '@testing-library/jest-dom'
import {BrowserRouter} from 'react-router-dom';

describe('<ActionButtons />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <ActionButtons mode='create' type='application' values={{}}/>,
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
