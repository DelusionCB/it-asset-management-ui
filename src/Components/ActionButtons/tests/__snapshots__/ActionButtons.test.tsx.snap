// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButtons /> Snapshot renders correctly 1`] = `
[
  <div>
    <button
      className="action-button btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      application.edit
    </button>
    <button
      className="action-button btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      application.create
    </button>
  </div>,
  ",",
]
`;
